addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -3412,6 +3412,19 @@ export class Wysiwyg extends Component {
             delete el.dataset.bgSrc;
         } else {
             el.setAttribute('src', newAttachmentSrc);
+            // Also update carousel thumbnail.
+            const carouselEl = el.closest(".carousel");
+            if (carouselEl) {
+                const carouselInnerEl = el.closest(".carousel-inner");
+                const carouselItemEl = el.closest(".carousel-item");
+                if (carouselInnerEl && carouselItemEl) {
+                    const imageIndex = [...carouselInnerEl.children].indexOf(carouselItemEl);
+                    const miniatureEl = carouselEl.querySelector(`.carousel-indicators [data-bs-slide-to="${imageIndex}"]`);
+                    if (miniatureEl) {
+                        miniatureEl.style.backgroundImage = `url(${newAttachmentSrc})`;
```

Comment:
As we are targeting `Boostrap` classes and not specifically the `s_image_gallery` snippet, I would make it more robust by checking if there is a "background image" to see if we can do this. In fact, BS carousel indicators do not systematically have a bg image. (e.g. the "Carousel" snippet, wich is not affected by this maybe because the DOM does not have exactly the same structure or the same order of the elements)